### PR TITLE
#562 Put constructor logic for subst_config in python

### PR DESCRIPTION
**Pull request by mortalisk**
What has been done:
 - Have created basic logic for constructing subst_config in python. #562 
 - Create comparison `__eq__` #563 
 - Some tests for both

### CODE REVIEW

**Comment by ghost** [comment]
Can one of the admins verify this patch?

File: python/res/enkf/subst_config.py [hunk 1]
```diff
@@ -47,3 +107,22 @@ def subst_list(self):
 
     def free(self):
         self._free()
+
+    def __eq__(self, other):
+        list1 = self.subst_list
+        list2 = other.subst_list
+        if len(list1.keys()) != len(list2.keys()):
+            print "length is different"
```
Comment (ManInFez) [inline python/res/enkf/subst_config.py]: print is not python 3 compliant, and is also not necessary

File: python/res/enkf/subst_config.py [hunk 2]
```diff
@@ -47,3 +107,22 @@ def subst_list(self):
...
+            return False
+        for key in list1.keys():
+            val1 = list1.get(key)
+            val2 = list2.get(key)
+            if val1 != val2:
+                print "key {} is different, values: {}, {}".format(key, val1, val2)
```
Comment (ManInFez) [inline python/res/enkf/subst_config.py]: print is not python 3 compliant, and is also not necessary

File: python/res/enkf/subst_config.py [hunk 3]
```diff
@@ -14,22 +14,82 @@
...
     _alloc          = ResPrototype("void* subst_config_alloc(config_content)", bind=False)
+    _alloc_full     = ResPrototype("void* subst_config_alloc_full(subst_list)", bind=False)
     _free           = ResPrototype("void  subst_config_free(subst_config)")
     _get_subst_list = ResPrototype("subst_list_ref subst_config_get_subst_list( subst_config )")
+    _get_num_cpu    = EclPrototype("int ecl_util_get_num_cpu(char*)", bind = False)
 
-    def __init__(self, config_content):
-        c_ptr = self._alloc(config_content)
+    def __init__(self, config_content=None, config_dict=None):
+        if not config_content and config_dict:
```
Comment (ManInFez) [inline python/res/enkf/subst_config.py]: Be explicit with ```is None``` and ```is not None````
Also could be changed to  
```
if not ( (config_content is not None) ^ (config_dict is not None)):
    raise ValueError(...)

**Comment by ManInFez** [comment]
Looks good for the most part, but need some fixes to pass python 3 tests. Also, resolve conflicts with master branch.

File: python/res/enkf/subst_config.py [hunk 4]
```diff
@@ -47,3 +107,22 @@ def subst_list(self):
 
     def free(self):
         self._free()
+
```
Comment (ManInFez) [inline python/res/enkf/subst_config.py:111]: Also include the __ne__ operator

File: python/tests/res/enkf/test_subst_config.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            subst_config1 = res_config.subst_config
+            subst_config2 = SubstConfig(config_dict=self.set_key(ConfigKeys.CONFIG_DIRECTORY, cwd))
+
+            self.assertEqual(
+                subst_config1,
+                subst_config2,
+                str(subst_config1) + "\n\nis not equal to:\n\n" + str(subst_config2))
+
+    def test_complete_config_reads_correct_values(self):
+        print os.getcwd()
```
Comment (ManInFez) [inline python/tests/res/enkf/test_subst_config.py]: Still one python 3 compliant print statement to go :)

**Comment by ManInFez** [comment]
Squash failing commits before merge

**Review by ManInFez — APPROVED** [review]
LGTM!

**Pull request merged**